stl/inc/format

```diff
@@ -3682,7 +3682,7 @@ public:
     template <class _Ty>
         requires convertible_to<const _Ty&, basic_string_view<_CharT>>
     consteval basic_format_string(const _Ty& _Str_val) : _Str(_Str_val) {
-        if (_Is_execution_charset_self_synchronizing()) {
+        if constexpr (!is_same_v<_CharT, char> || _Is_execution_charset_self_synchronizing()) {

```

Comment:
why not `is_same_v<_CharT, wchar_t>`? That seems more direct to me.

Reply:
My opinion was that `!is_same_v<_CharT, char>` would "automatically" support `charN_t` if formatting for them gets supported in the future.
> (or is not `char`, if formatting for `charN_t` is supported in the future)

But this reason may be not stong enough.